File: ordenarArquivos/src/main/java/ordenarArquivos/model/Manga.java
package ordenarArquivos.model;

import java.time.LocalDateTime;
import java.util.ArrayList;

public class Manga {

	private Long id;
	private String nome;
	private String volume;
	private String capitulo;
	private String arquivo;
	private String capitulos;
	private Integer quantidade;
	private LocalDateTime atualizacao = LocalDateTime.now();
	private ArrayList<Caminhos> caminhos = new ArrayList<Caminhos>();

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getVolume() {
		return volume;
	}

	public void setVolume(String volume) {
		this.volume = volume;
	}

	public String getCapitulo() {
		return capitulo;
	}

	public void setCapitulo(String capitulo) {
		this.capitulo = capitulo;
	}

	public String getArquivo() {
		return arquivo;
	}

	public void setArquivo(String arquivo) {
		this.arquivo = arquivo;
	}

	public Integer getQuantidade() {
		return quantidade;
	}

	public void setQuantidade(Integer quantidade) {
		this.quantidade = quantidade;
	}

	public String getCapitulos() {
		return capitulos;
	}

	public void setCapitulos(String capitulos) {
		this.capitulos = capitulos;
	}

	public ArrayList<Caminhos> getCaminhos() {
		return caminhos;
	}

	public void setCaminhos(ArrayList<Caminhos> caminhos) {
		this.caminhos = caminhos;
	}

	public void addCaminhos(Caminhos caminhos) {
		this.caminhos.add(caminhos);
	}

	public LocalDateTime getAtualizacao() {
		return atualizacao;
	}

	public void setAtualizacao(LocalDateTime atualizacao) {
		this.atualizacao = atualizacao;
	}

	public Manga(Long id, String nome, String volume, String capitulo, String arquivo, Integer quantidade,
			String capitulos) {
		this.id = id;
		this.nome = nome;
		this.volume = volume;
		this.capitulo = capitulo;
		this.arquivo = arquivo;
		this.quantidade = quantidade;
		this.capitulos = capitulos;
	}

	public Manga(Long id, String nome, String volume, String capitulo, String arquivo, Integer quantidade,
			String capitulos, LocalDateTime atualizacao) {
		this.id = id;
		this.nome = nome;
		this.volume = volume;
		this.capitulo = capitulo;
		this.arquivo = arquivo;
		this.atualizacao = atualizacao;
		this.quantidade = quantidade;
		this.capitulos = capitulos;
	}

	@Override
	public String toString() {
		return "Manga [id=" + id + ", nome=" + nome + ", volume=" + volume + ", capitulo=" + capitulo + ", arquivo="
				+ arquivo + "]";
	}

}

File: OrdenarArquivos/src/main/kotlin/com/fenix/ordenararquivos/model/Manga.kt
package com.fenix.ordenararquivos.model

import java.time.LocalDateTime

data class Manga(
    var id: Long = 0,
    var nome: String = "",
    var volume: String = "",
    var capitulo: String = "",
    var arquivo: String = "",
    var capitulos: String = "",
    var quantidade: Int = 0,
    var atualizacao: LocalDateTime = LocalDateTime.now(),
    var caminhos: MutableList<Caminhos> = arrayListOf()
) {
    constructor(
        id: Long,
        nome: String,
        volume: String,
        capitulo: String,
        arquivo: String,
        quantidade: Int,
        capitulos: String,
        atualizacao: LocalDateTime
    ) : this(
        id,
        nome,
        volume,
        capitulo,
        arquivo,
        capitulos,
        quantidade,
        atualizacao
    ) {

    }

    fun addCaminhos(caminhos: Caminhos) {
        this.caminhos.add(caminhos)
    }

    override fun toString(): String {
        return ("Manga [id=" + id + ", nome=" + nome + ", volume=" + volume + ", capitulo=" + capitulo + ", arquivo="
                + arquivo + "]")
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Manga

        if (nome != other.nome) return false
        if (volume != other.volume) return false
        if (capitulo != other.capitulo) return false
        if (arquivo != other.arquivo) return false
        if (capitulos != other.capitulos) return false

        return true
    }

    override fun hashCode(): Int {
        var result = nome.hashCode()
        result = 31 * result + volume.hashCode()
        result = 31 * result + capitulo.hashCode()
        result = 31 * result + arquivo.hashCode()
        result = 31 * result + capitulos.hashCode()
        return result
    }

}